add error notification if /restore API fails

Plugin to use: 

https://www.npmjs.com/package/react-notifications

Is everything okay with this library?

It seems to be deprecated. Their documentation still uses class components only.

I installed it by running "sudo npm install --save react-notifications" and then tried to import it with the statement: "import { NotificationManager} from 'react-notifications'" but then I got this error: "Could not find a declaration file for module 'react-notifications'."

Then I tried installing the types by running "npm i --save-dev @types/react-notifications" and got another error: "npm ERR! 404 Not Found - GET https://registry.npmjs.org/@types%2freact-notifications - Not found"

One possible solution would be to use another library, such as "react-hot-toast". Would that be ok?

--- a/app/components/UploadComponent.tsx
+++ b/app/components/UploadComponent.tsx
@@ -14,6 +14,7 @@ import { Button } from 'greenhouse-react-ui'
 import { useMediaQuery } from '@react-hook/media-query'
 import Link from 'next/link'
 import { RotatingLines } from 'react-loader-spinner'
+import toast, { Toaster } from 'react-hot-toast'
 
 if (!process.env.NEXT_PUBLIC_UPLOAD_IO_API_KEY)
   throw new Error('UPLOAD_IO_API_KEY is not set')
@@ -97,6 +98,7 @@ export default function UploadComponent() {
       setRestoredImageUrl(data.restoredImageUrl)
       setLoading(false)
     } catch (error) {
+      toast.error('Something went wrong. Please check your internet connection')
       setLoading(false)
       console.error(error)
     }
@@ -119,110 +121,113 @@ export default function UploadComponent() {
   }
 
   return (
-    <div className="flex items-center flex-col mx-auto md:mt-12 w-full">
-      <RotatingLines visible={loading} />
-      {imageUrl && restoredImageUrl && (
-        <>
-          <div className="flex gap-4 md:gap-10 justify-center items-center">
-            <span>Split view</span>
-            <Toggle
-              enabled={sidebyside}
-              setEnabled={(val) => setSidebyside(val)}
-            />
-            <span>Slider view</span>
-          </div>
-          {!sidebyside && (
-            <CompareSlider
-              original={imageUrl}
-              restored={restoredImageUrl}
-              classNames={`w-[${
-                matches ? 400 : 300
-              }px] flex justify-center items-center`}
-              // TODO: Fix ts error
-              // @ts-ignore
-              portrait={true}
-            />
-          )}
-        </>
-      )}
-      {!imageUrl && (
-        <UploadDropzone
-          uploader={uploader}
-          options={uploaderOptions}
-          onUpdate={(files) => {
-            if (files.length === 0) {
-              console.log('No files selected.')
-            } else {
-              console.log('File selected: ', files[0].fileUrl)
-              setImageName(files[0].originalFile.originalFileName)
-              setImageUrl(files[0].fileUrl)
-              restoreImage(files[0].fileUrl)
-            }
-          }}
-          // onComplete={(files) => alert(files.map((x) => x.fileUrl).join('\n'))}
-          width="600px"
-          height="375px"
-          className="mx-auto"
-        />
-      )}
-      <div className="flex flex-col md:flex-row md:gap-10 justify-center align-center">
-        {sidebyside && (
+    <>
+      <Toaster />
+      <div className="flex items-center flex-col mx-auto md:mt-12 w-full">
+        <RotatingLines visible={loading} />
+        {imageUrl && restoredImageUrl && (
           <>
-            {/* https://upcdn.io/12a1yJB/raw/uploads/2023/06/04/PASSPORT_PHOTO-5wmg.jpg */}
-            {imageUrl && (
-              <div className="flex flex-col">
-                <Image
-                  src={imageUrl}
-                  alt="Main image"
-                  width={matches ? 400 : 300}
-                  height={matches ? 400 : 300}
-                />
-
-                {!loading && (
-                  <Button
-                    onClick={resetFields}
-                    className="mt-5 md:mt-10"
-                    size={matches ? 'medium' : 'large'}
-                    layout="outline"
-                  >
-                    Upload another image
-                  </Button>
-                )}
-              </div>
-            )}
-            {restoredImageUrl && (
-              // <a href={restoredImageUrl} target="_blank" rel="noreferrer">
-              <div className="flex flex-col mt-5 md:mt-0">
-                <Link href={restoredImageUrl} target="_blank">
-                  <Image
-                    src={restoredImageUrl}
-                    alt="Restored image"
-                    width={matches ? 400 : 300}
-                    height={matches ? 400 : 300}
-                    onLoad={() => setRestoredImageLoaded(true)}
-                  />
-                </Link>
-                {/* TODO: Fix sonarlint warning */}
-                {restoredImageLoaded && (
-                  <Button
-                    onClick={downloadRestoredImg}
-                    className="mt-5 md:mt-10"
-                    layout="outline"
-                    size={matches ? 'medium' : 'large'}
-                  >
-                    {downloading
-                      ? 'Downloading Restored Image'
-                      : 'Download Restored Image'}
-                  </Button>
-                )}
-                {/* https://replicate.delivery/pbxt/FeUR3TUZGM0GFSEx7FgWbkwWXSkwEP3PeMS99emefZKdDETIC/output.png */}
-              </div>
-
-              // </a>
+            <div className="flex gap-4 md:gap-10 justify-center items-center">
+              <span>Split view</span>
+              <Toggle
+                enabled={sidebyside}
+                setEnabled={(val) => setSidebyside(val)}
+              />
+              <span>Slider view</span>
+            </div>
+            {!sidebyside && (
+              <CompareSlider
+                original={imageUrl}
+                restored={restoredImageUrl}
+                classNames={`w-[${
+                  matches ? 400 : 300
+                }px] flex justify-center items-center`}
+                // TODO: Fix ts error
+                // @ts-ignore
+                portrait={true}
+              />
             )}
           </>
         )}
+        {!imageUrl && (
+          <UploadDropzone
+            uploader={uploader}
+            options={uploaderOptions}
+            onUpdate={(files) => {
+              if (files.length === 0) {
+                console.log('No files selected.')
+              } else {
+                console.log('File selected: ', files[0].fileUrl)
+                setImageName(files[0].originalFile.originalFileName)
+                setImageUrl(files[0].fileUrl)
+                restoreImage(files[0].fileUrl)
+              }
+            }}
+            // onComplete={(files) => alert(files.map((x) => x.fileUrl).join('\n'))}
+            width="600px"
+            height="375px"
+            className="mx-auto"
+          />
+        )}
+        <div className="flex flex-col md:flex-row md:gap-10 justify-center align-center">
+          {sidebyside && (
+            <>
+              {/* https://upcdn.io/12a1yJB/raw/uploads/2023/06/04/PASSPORT_PHOTO-5wmg.jpg */}
+              {imageUrl && (
+                <div className="flex flex-col">
+                  <Image
+                    src={imageUrl}
+                    alt="Main image"
+                    width={matches ? 400 : 300}
+                    height={matches ? 400 : 300}
+                  />
+
+                  {!loading && (
+                    <Button
+                      onClick={resetFields}
+                      className="mt-5 md:mt-10"
+                      size={matches ? 'medium' : 'large'}
+                      layout="outline"
+                    >
+                      Upload another image
+                    </Button>
+                  )}
+                </div>
+              )}
+              {restoredImageUrl && (
+                // <a href={restoredImageUrl} target="_blank" rel="noreferrer">
+                <div className="flex flex-col mt-5 md:mt-0">
+                  <Link href={restoredImageUrl} target="_blank">
+                    <Image
+                      src={restoredImageUrl}
+                      alt="Restored image"
+                      width={matches ? 400 : 300}
+                      height={matches ? 400 : 300}
+                      onLoad={() => setRestoredImageLoaded(true)}
+                    />
+                  </Link>
+                  {/* TODO: Fix sonarlint warning */}
+                  {restoredImageLoaded && (
+                    <Button
+                      onClick={downloadRestoredImg}
+                      className="mt-5 md:mt-10"
+                      layout="outline"
+                      size={matches ? 'medium' : 'large'}
+                    >
+                      {downloading
+                        ? 'Downloading Restored Image'
+                        : 'Download Restored Image'}
+                    </Button>
+                  )}
+                  {/* https://replicate.delivery/pbxt/FeUR3TUZGM0GFSEx7FgWbkwWXSkwEP3PeMS99emefZKdDETIC/output.png */}
+                </div>
+
+                // </a>
+              )}
+            </>
+          )}
+        </div>
       </div>
-    </div>
+    </>
   )
 }

--- a/package.json
+++ b/package.json
@@ -29,6 +29,7 @@
     "react": "18.2.0",
     "react-compare-slider": "2.2.0",
     "react-dom": "18.2.0",
+    "react-hot-toast": "^2.4.1",
     "react-icons": "4.9.0",
     "react-loader-spinner": "^5.3.4",
     "react-uploader": "^3.0.2",

